### PR TITLE
Fixes #25188: Improve system-updates module

### PR DESCRIPTION
**Pull request by amousset**
https://issues.rudder.io/issues/25188

### CODE REVIEW

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

File: policies/module-types/system-updates/README.md [hunk 1]
```diff
@@ -1,8 +1,75 @@
...
+
+#### APT
+
+We decided to use the `libapt` C++ library through Rust bindings instead of the CLI interface. This has several advantages:
+
+* Allow doing security-only upgrades sanely (like `unattended-upgrade`, the only official way, does), without disabling core repositories.
+* Share the APT cache for the best performance and always up-to-date cache.
+
+The drawbacks:
+* Makes the build a bit mure cumbersome.
```
Comment (Fdall) [inline policies/module-types/system-updates/README.md:59]:
```suggestion
* Makes the build a bit more cumbersome.
```

**Comment by amousset** [comment]
PR updated with a new commit

File: policies/module-types/system-updates/src/campaign.rs [hunk 2]
```diff
@@ -1,210 +1,202 @@
...
-    let now: DateTime<Utc> = Utc::now();
+    let schedule_datetime = match schedule {
+        FullSchedule::Immediate => Utc::now(),
+        FullSchedule::Scheduled(ref s) => {
+            scheduler::splayed_start(s.start, s.end, s.agent_frequency, s.node_id.as_str())?
+        }
+    };
+    let already_scheduled = db.schedule_event(&p.event_id, &p.campaign_name, schedule_datetime)?;
 
     // Update should have start/have started already
```
Comment (Fdall) [inline policies/module-types/system-updates/src/campaign.rs:58]:
```suggestion
    // Update should have started already
```

File: policies/module-types/system-updates/src/cli.rs [hunk 3]
```diff
@@ -0,0 +1,158 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// SPDX-FileCopyrightText: 2024 Normation SAS
+#![allow(unused_imports)]
```
Comment (Fdall) [inline policies/module-types/system-updates/src/cli.rs:3]: why ?

File: policies/module-types/system-updates/src/cli.rs [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+}
+
+#[derive(Debug, Options)]
+struct RunOpts {
+    #[options(help = "print help message")]
+    help: bool,
+    #[options(help = "directory where the database is stored")]
+    state_dir: Option<PathBuf>,
+    #[options(long = "dry-run", help = "do not apply changes")]
+    dry_run: bool,
```
Comment (Fdall) [inline policies/module-types/system-updates/src/cli.rs:55]: I could not find where this is used

**Review by Fdall — APPROVED** [review]

File: policies/module-types/system-updates/src/cli.rs (hunk 4 above)
Comment (amousset) [reply policies/module-types/system-updates/src/cli.rs:55]: it is not :)

**Pull request merged**